### DISCUSSION TITLE
Axios Post Request的data 屬性應該傳入什麼類型的資料呢？

### DISCUSSION THREAD

**Discussion by TehoChang** [Lidemy/forum#251] [其他問題]
我有一個axios函數如下
```jsx
function createOrder({custkey, token, payload}){
    return myAxios({
      url:"/create.order",
      method:'POST',
      params:{
        custkey,
        token
      },
      data:payload
    })
}
```
API要求的Payload是一個陣列，資料內容大概如下
```jsx
const payload =[
  {
    "projectcode": "jdorder",
    "shipref": "DT20222H000000045P_0224_001",
    "secondref": "",
    "routecode": "",
    "carriercode": "",
    "waypoints": [
      {
          "pointtype": "P",
          "companyname": undefined,
        }
    ]
  }
]
```
當我將payLoad中所有屬性的雙引號都去除，然後發送請求，API回報數據格式錯誤
將payload進行轉換
`payload = JSON.parse(JSON.stringify(payload))` >> 發送請求，回報數據格式錯誤
`payload = JSON.stringify(payload)` >> 發送請求，回報數據格式錯誤

看似只有當payload的所有屬性都有雙引號時，請求才會成功
可是當我使用postman進行測試，將body設置成raw/text，發送請求，api成功執行
![截圖 2022-02-24 上午1 06 28](https://user-images.githubusercontent.com/61130008/155369872-155e3231-403b-4091-9003-a649ce7d0f3e.png)

無法理解是axios的data_field只接受特定類型的資料，還是API後端的php程式碼有什麼設定？

懇請好心人指教，感謝！

**Comment by aszx87410** [comment]
照理來說直接用 axios 打應該要過，你可以問一下後端對於資料格式會做什麼檢查，什麼狀況下會出錯，或跟他要程式碼來看一下
